zboxcore/sdk/registry_file_manager.go

```diff
@@ -0,0 +1,203 @@
+package sdk
+
+import (
+	"fmt"
+	"github.com/0chain/errors"
+	"github.com/0chain/gosdk/core/common"
+	"github.com/0chain/gosdk/core/transaction"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+)
+
+// allocationFileStorer defines functions expected from Allocation for storing file.
+// This interface enables easy mocking for UT purposes.
+type allocationFileStorer interface {
+	ListDir(path string) (*ListResult, error)
+	DownloadFile(localPath string, remotePath string, status StatusCallback) error
+	StartChunkedUpload(workdir, localPath string, remotePath string, status StatusCallback, isUpdate bool, isRepair bool, thumbnailPath string, encryption bool) error
+}
+
+var _ allocationFileStorer = &Allocation{}
+
+type registryFileManager interface {
+	Update(data []byte) error
+	Get() ([]byte, common.Timestamp, error)
+	GetLastUpdateTimestamp() (common.Timestamp, error)
+}
+
+// registryFileStore manages the storing of registry file on allocation.
+type registryFileStore struct {
+	registryFilePath string
+	allocation       *Allocation
+	fileStorer       allocationFileStorer
+}
+
+var _ registryFileManager = &registryFileStore{}
+
+func newRegistryFileManager(a *Allocation, registryFilePath string) *registryFileStore {
+	return &registryFileStore{
+		registryFilePath: registryFilePath,
+		allocation:       a,
+		fileStorer:       a,
+	}
+}
+
+// Update writes the registry file with data provided.
+// This overwrites the current copy stored.
+func (a *registryFileStore) Update(data []byte) error {
+	rf, err := a.getRegistryFile()
+	if err != nil {
+		return errors.New("update_registry_file_failed", "Failed to check existence of registry file: "+err.Error())
```

Comment:
Do not capitalize error message, so please use "failed to check exitence of registry file..."